clvm/SExp.py

```diff
@@ -16,27 +17,27 @@
 
 CastableType = typing.Union[
     "SExp",
-    "CLVMObject",
+    CLVMObjectLike,
     bytes,
     str,
     int,
     None,
     list,
-    typing.Tuple[typing.Any, typing.Any],
+    typing.Tuple["CastableType", ...],
 ]
 
 
 NULL = b""
 
 
-def looks_like_clvm_object(o: typing.Any) -> bool:
+def looks_like_clvm_object(o: typing.Any) -> typing_extensions.TypeGuard[CLVMObjectLike]:
     d = dir(o)
     return "atom" in d and "pair" in d
 
 
 # this function recognizes some common types and turns them into plain bytes,
 def convert_atom_to_bytes(
-    v: typing.Union[bytes, str, int, None, list],
+    v: typing.Union[bytes, str, int, None, typing.List, typing.SupportsBytes],
```

Comment:
This is a bit tricky. `v` can be an empty list, but no other `list` type is valid. This is probably a mistake, and instead it should probably be handled in `.to` instead of here.

Reply:
So maybe removing `list` from here is the best idea. This isn't intended to be a public function.

Reply:
I'd say we try taking out the `[]` case and see if anything breaks. I suspect no one is using this function besides `.to`.

Reply:
I literally just this evening removed a comment about the empty-list-only aspect...  `:]`

Checking the change at https://github.com/Chia-Network/clvm/pull/143.

Reply:
quex knew how to do this.  It's a bit odd, but works.  If we want to continue with #143, that's ok, but I think this lets us separate it from this PR while retaining complete and correct hinting here.

```suggestion
    v: typing.Union[bytes, str, int, None, typing.List[typing_extensions.Never], typing.SupportsBytes],
```

https://mypy-play.net/?mypy=1.7.1&python=3.11&gist=c729cdd00474aae37ce40fb6325386b2
```python
from typing import List, Never

def f(el: List[Never]) -> None:
    return

f([])
f(["a"])
```

```
main.py:7: error: Argument 1 to "f" has incompatible type "list[str]"; expected "list[NoReturn]"  [arg-type]
Found 1 error in 1 file (checked 1 source file)
```